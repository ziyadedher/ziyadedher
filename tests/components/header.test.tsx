/**
 * @jest-environment jsdom
 */

// eslint-disable-next-line import/no-unassigned-import -- @testing-library/jest-dom doesn't need to be bound.
import "@testing-library/jest-dom";
import { render } from "@testing-library/react";

import Header from "../../src/components/header";

describe("header", () => {
  test("renders", () => {
    const result = render(<Header />);
    expect(result).toMatchSnapshot();
  });
});
