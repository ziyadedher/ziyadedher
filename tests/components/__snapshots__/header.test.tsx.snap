// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header renders 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="flex flex-col items-center py-12 text-center"
      >
        <div
          class="text-2xl font-light uppercase hover:opacity-70 transition-all"
        >
          <a
            href="/"
          >
            Ziyad Edher
          </a>
        </div>
        <div
          class="flex flex-row pt-2 space-x-2"
        >
          <span
            class="hover:opacity-50 transition-all"
          >
            <a
              aria-label="Twitter"
              href="https://twitter.com/ziyadedher"
              rel="noreferrer"
              target="_blank"
            >
              <svg
                fill="currentColor"
                height="24"
                viewBox="0 0 256 256"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <rect
                  fill="none"
                  height="256"
                  width="256"
                />
                <path
                  d="M128,88.00288a40.00668,40.00668,0,0,1,76.67148-16.00327L240,72l-32.26239,32.2625A128.00746,128.00746,0,0,1,80,224c-32,0-40-12-40-12s32-12,48-36c0,0-64-32-48-120,0,0,40,40,87.9862,48Z"
                  fill="none"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="12"
                />
              </svg>
            </a>
          </span>
          <span
            class="hover:opacity-50 transition-all"
          >
            <a
              aria-label="LinkedIn"
              href="https://linkedin.com/in/ziyad-edher"
              rel="noreferrer"
              target="_blank"
            >
              <svg
                fill="currentColor"
                height="24"
                viewBox="0 0 256 256"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <rect
                  fill="none"
                  height="256"
                  width="256"
                />
                <rect
                  fill="none"
                  height="184"
                  rx="8"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="12"
                  width="184"
                  x="36"
                  y="36"
                />
                <line
                  fill="none"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="12"
                  x1="120"
                  x2="120"
                  y1="112.00094"
                  y2="176.00094"
                />
                <line
                  fill="none"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="12"
                  x1="88"
                  x2="88"
                  y1="112.00094"
                  y2="176.00094"
                />
                <path
                  d="M120,140.00094a28,28,0,1,1,56,0v36"
                  fill="none"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="12"
                />
                <circle
                  cx="88"
                  cy="79.99998"
                  r="10"
                />
              </svg>
            </a>
          </span>
          <span
            class="hover:opacity-50 transition-all"
          >
            <a
              aria-label="GitHub"
              href="https://github.com/ziyadedher"
              rel="noreferrer"
              target="_blank"
            >
              <svg
                fill="currentColor"
                height="24"
                viewBox="0 0 256 256"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <rect
                  fill="none"
                  height="256"
                  width="256"
                />
                <path
                  d="M84,240a24,24,0,0,0,24-24V168"
                  fill="none"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="12"
                />
                <path
                  d="M172,240a24,24,0,0,1-24-24V168"
                  fill="none"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="12"
                />
                <path
                  d="M152,168h16a24,24,0,0,1,24,24v8a24,24,0,0,0,24,24"
                  fill="none"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="12"
                />
                <path
                  d="M104,168H88a24,24,0,0,0-24,24v8a24,24,0,0,1-24,24"
                  fill="none"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="12"
                />
                <path
                  d="M111.825,63.99934A51.9599,51.9599,0,0,0,68,40a51.90058,51.90058,0,0,0-3.48841,44.7036A49.25789,49.25789,0,0,0,56,112v8a48,48,0,0,0,48,48h48a48,48,0,0,0,48-48v-8a49.25769,49.25769,0,0,0-8.5116-27.29639A51.90061,51.90061,0,0,0,188,40a51.95992,51.95992,0,0,0-43.82535,23.99983Z"
                  fill="none"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="12"
                />
              </svg>
            </a>
          </span>
          <span
            class="hover:opacity-50 transition-all"
          >
            <a
              aria-label="Spotify"
              href="https://open.spotify.com/user/c9n8y6h3ssu8hg811rfqt4tvrr"
              rel="noreferrer"
              target="_blank"
            >
              <svg
                fill="currentColor"
                height="24"
                viewBox="0 0 256 256"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <rect
                  fill="none"
                  height="256"
                  width="256"
                />
                <circle
                  cx="128"
                  cy="128"
                  fill="none"
                  r="96"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="12"
                />
                <g>
                  <path
                    d="M179.1333,108.32931a112.19069,112.19069,0,0,0-102.3584.04859"
                    fill="none"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="12"
                  />
                  <path
                    d="M164.29541,136.71457a79.94058,79.94058,0,0,0-72.68359.04736"
                    fill="none"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="12"
                  />
                  <path
                    d="M149.47217,165.07248a47.97816,47.97816,0,0,0-43.03662.04736"
                    fill="none"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="12"
                  />
                </g>
              </svg>
            </a>
          </span>
          <span
            class="hover:opacity-50 transition-all"
          >
            <a
              aria-label="Facebook"
              href="https://facebook.com/ziyadedher"
              rel="noreferrer"
              target="_blank"
            >
              <svg
                fill="currentColor"
                height="24"
                viewBox="0 0 256 256"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <rect
                  fill="none"
                  height="256"
                  width="256"
                />
                <circle
                  cx="128"
                  cy="128"
                  fill="none"
                  r="96"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="12"
                />
                <path
                  d="M168,88.00094H152a24,24,0,0,0-24,24v112"
                  fill="none"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="12"
                />
                <line
                  fill="none"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="12"
                  x1="96"
                  x2="160"
                  y1="144.00094"
                  y2="144.00094"
                />
              </svg>
            </a>
          </span>
          <span
            class="hover:opacity-50 transition-all"
          >
            <a
              aria-label="Instagram"
              href="https://instagram.com/ziyadedher"
              rel="noreferrer"
              target="_blank"
            >
              <svg
                fill="currentColor"
                height="24"
                viewBox="0 0 256 256"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <rect
                  fill="none"
                  height="256"
                  width="256"
                />
                <circle
                  cx="128"
                  cy="128"
                  fill="none"
                  r="40"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="12"
                />
                <rect
                  fill="none"
                  height="184"
                  rx="48"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="12"
                  width="184"
                  x="36"
                  y="36"
                />
                <circle
                  cx="180"
                  cy="75.99998"
                  r="10"
                />
              </svg>
            </a>
          </span>
        </div>
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      class="flex flex-col items-center py-12 text-center"
    >
      <div
        class="text-2xl font-light uppercase hover:opacity-70 transition-all"
      >
        <a
          href="/"
        >
          Ziyad Edher
        </a>
      </div>
      <div
        class="flex flex-row pt-2 space-x-2"
      >
        <span
          class="hover:opacity-50 transition-all"
        >
          <a
            aria-label="Twitter"
            href="https://twitter.com/ziyadedher"
            rel="noreferrer"
            target="_blank"
          >
            <svg
              fill="currentColor"
              height="24"
              viewBox="0 0 256 256"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <rect
                fill="none"
                height="256"
                width="256"
              />
              <path
                d="M128,88.00288a40.00668,40.00668,0,0,1,76.67148-16.00327L240,72l-32.26239,32.2625A128.00746,128.00746,0,0,1,80,224c-32,0-40-12-40-12s32-12,48-36c0,0-64-32-48-120,0,0,40,40,87.9862,48Z"
                fill="none"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="12"
              />
            </svg>
          </a>
        </span>
        <span
          class="hover:opacity-50 transition-all"
        >
          <a
            aria-label="LinkedIn"
            href="https://linkedin.com/in/ziyad-edher"
            rel="noreferrer"
            target="_blank"
          >
            <svg
              fill="currentColor"
              height="24"
              viewBox="0 0 256 256"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <rect
                fill="none"
                height="256"
                width="256"
              />
              <rect
                fill="none"
                height="184"
                rx="8"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="12"
                width="184"
                x="36"
                y="36"
              />
              <line
                fill="none"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="12"
                x1="120"
                x2="120"
                y1="112.00094"
                y2="176.00094"
              />
              <line
                fill="none"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="12"
                x1="88"
                x2="88"
                y1="112.00094"
                y2="176.00094"
              />
              <path
                d="M120,140.00094a28,28,0,1,1,56,0v36"
                fill="none"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="12"
              />
              <circle
                cx="88"
                cy="79.99998"
                r="10"
              />
            </svg>
          </a>
        </span>
        <span
          class="hover:opacity-50 transition-all"
        >
          <a
            aria-label="GitHub"
            href="https://github.com/ziyadedher"
            rel="noreferrer"
            target="_blank"
          >
            <svg
              fill="currentColor"
              height="24"
              viewBox="0 0 256 256"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <rect
                fill="none"
                height="256"
                width="256"
              />
              <path
                d="M84,240a24,24,0,0,0,24-24V168"
                fill="none"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="12"
              />
              <path
                d="M172,240a24,24,0,0,1-24-24V168"
                fill="none"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="12"
              />
              <path
                d="M152,168h16a24,24,0,0,1,24,24v8a24,24,0,0,0,24,24"
                fill="none"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="12"
              />
              <path
                d="M104,168H88a24,24,0,0,0-24,24v8a24,24,0,0,1-24,24"
                fill="none"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="12"
              />
              <path
                d="M111.825,63.99934A51.9599,51.9599,0,0,0,68,40a51.90058,51.90058,0,0,0-3.48841,44.7036A49.25789,49.25789,0,0,0,56,112v8a48,48,0,0,0,48,48h48a48,48,0,0,0,48-48v-8a49.25769,49.25769,0,0,0-8.5116-27.29639A51.90061,51.90061,0,0,0,188,40a51.95992,51.95992,0,0,0-43.82535,23.99983Z"
                fill="none"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="12"
              />
            </svg>
          </a>
        </span>
        <span
          class="hover:opacity-50 transition-all"
        >
          <a
            aria-label="Spotify"
            href="https://open.spotify.com/user/c9n8y6h3ssu8hg811rfqt4tvrr"
            rel="noreferrer"
            target="_blank"
          >
            <svg
              fill="currentColor"
              height="24"
              viewBox="0 0 256 256"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <rect
                fill="none"
                height="256"
                width="256"
              />
              <circle
                cx="128"
                cy="128"
                fill="none"
                r="96"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="12"
              />
              <g>
                <path
                  d="M179.1333,108.32931a112.19069,112.19069,0,0,0-102.3584.04859"
                  fill="none"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="12"
                />
                <path
                  d="M164.29541,136.71457a79.94058,79.94058,0,0,0-72.68359.04736"
                  fill="none"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="12"
                />
                <path
                  d="M149.47217,165.07248a47.97816,47.97816,0,0,0-43.03662.04736"
                  fill="none"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="12"
                />
              </g>
            </svg>
          </a>
        </span>
        <span
          class="hover:opacity-50 transition-all"
        >
          <a
            aria-label="Facebook"
            href="https://facebook.com/ziyadedher"
            rel="noreferrer"
            target="_blank"
          >
            <svg
              fill="currentColor"
              height="24"
              viewBox="0 0 256 256"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <rect
                fill="none"
                height="256"
                width="256"
              />
              <circle
                cx="128"
                cy="128"
                fill="none"
                r="96"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="12"
              />
              <path
                d="M168,88.00094H152a24,24,0,0,0-24,24v112"
                fill="none"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="12"
              />
              <line
                fill="none"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="12"
                x1="96"
                x2="160"
                y1="144.00094"
                y2="144.00094"
              />
            </svg>
          </a>
        </span>
        <span
          class="hover:opacity-50 transition-all"
        >
          <a
            aria-label="Instagram"
            href="https://instagram.com/ziyadedher"
            rel="noreferrer"
            target="_blank"
          >
            <svg
              fill="currentColor"
              height="24"
              viewBox="0 0 256 256"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <rect
                fill="none"
                height="256"
                width="256"
              />
              <circle
                cx="128"
                cy="128"
                fill="none"
                r="40"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="12"
              />
              <rect
                fill="none"
                height="184"
                rx="48"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="12"
                width="184"
                x="36"
                y="36"
              />
              <circle
                cx="180"
                cy="75.99998"
                r="10"
              />
            </svg>
          </a>
        </span>
      </div>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
